trunion/views.py

```diff
@@ -47,3 +47,20 @@ def sign_app(request):
     fname = os.path.splitext(request.POST['file'].filename)[0]
     pkcs7 = crypto.sign_app(request.POST['file'].file.read())
     return {fname + '.rsa': b64encode(pkcs7)}
+
+
+signaddon = Service(name='sign_addon', path='/1.0/sign_addon',
+                    description="Addon signer")
+
+
+@signaddon.post(validators=valid_addon)
+def sign_addon(request):
+    if request.registry.settings['trunion.we_are_signing'] != 'addons':
+        raise HTTPUnsupportedMediaType()
+
+    pkcs7 = crypto.sign_addon(request.POST['addon_id'],
+                              request.POST['file'].file.read())
```

Comment:
this all looks really fragile if the client sends wrong values or forgets to send a value. No validation?